### ISSUE TITLE
Difficulty replicating shape bias result

### ISSUE THREAD

**Issue by ajaysub110** [paulgavrikov/diffusion-noise-cls-pytorch#1]
Hi Paul,

Thanks for implementing this paper!

I trained ResNet-50 for 90 epochs with diffusion noise using your code, in an effort to replicate the results you report in your README.

I get very similar validation accuracies, but very different shape bias scores as shown in the table below. The second and third row are the diffusion-noise-trained ResNets without and with noise added during inference. As you can see my shape bias scores are much lower than yours.

<img width="693" alt="image" src="https://github.com/paulgavrikov/diffusion-noise-cls-pytorch/assets/27354211/885cd0bf-0682-491a-bfab-8aade0549ed3">

To measure shape bias I'm using code from https://github.com/bethgelab/model-vs-human as follows:

```python
from modelvshuman import Evaluate
Evaluate()(['resnet50'], ['cue-conflict'])
Evaluate()(['diffusion_resnet50'], ['cue-conflict'])
Evaluate()(['diffusion_resnet50_wdiffnoise'], ['cue-conflict'])
```

While testing the diffusion ResNets I'm also making sure to apply the required transforms (normalization to mean 0 and std 1, resize to 256 and center crop 224).

Any idea what might be causing my results to be so different?

Thanks,
Ajay

**Comment by paulgavrikov** [comment]
Hey Ajay,

I assume you are not using my codebase? My initial problem was wrong normalization and I got a similar shape bias to you. 2 things you may want to check:
1. You don't want 0 mean, 1 std - you just want the values to be within [0, 1] range. 
2. Make sure that you do this step *before* you apply the noise, i.e., the outputs of the "noise" inputs will have an unpredictable range but that is actually what you want. 

Also, your shape bias is slightly off for the pre-trained ResNet-50 if this refers to the original torchvision checkpoint. It should be roughly 21%.
